onyo/lib/pseudokeys.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from functools import partial
+from typing import TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from typing import (
+        Callable,
+        Dict,
+    )
+    from onyo.lib.items import Item
+
+
+@dataclass()
+class PseudoKey:
+    """"Defines a pseudo-key implementation."""
+    description: str
+    """description of what that pseudo-key is delivering"""
+    implementation: Callable
+    """Called by an `Item` when a pseudo-keys value is still undefined.
+
+    This Callable is expected to have a single parameter of type `Item`.
+    """
+
+
+def delegate(self: Item, attribute: str, *args, **kwargs):
+    # This is to avoid circular imports ATM.
+    return self.__getattribute__(attribute)(*args, **kwargs)
+
+
+PSEUDO_KEYS: Dict[str, PseudoKey] = {
```

Comment:
Nit. And I'm not sure if it matters, but this list is not alphabeitized.

I understand they're sorted so that similar things are near each other. But we will be adding quite a few more keys. Eventually, alphabetized is going to be the only sane way to find anything.

Reply:
Right. Will do. It does matter (kinda), simply because the main reason I wanted this kind of mapping in the first place, is the single entry point for humans.